src/components/NewTodo/NewTodo.tsx

```diff
@@ -0,0 +1,62 @@
+import React, {
+  useState,
+  useContext,
+  useRef,
+  useEffect,
+} from 'react';
+import { useClickAway } from 'react-use';
+import { TodosContext } from '../../context/TodosContext';
+import { Todo } from '../../types/Todo';
+
+export const NewTodo = () => {
+  const [value, setValue] = useState('');
+  const { todos, updateTodos } = useContext(TodosContext);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const ref = useRef(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [inputRef]);
+
+  const reset = () => {
+    setValue('');
+  };
+
+  const addNewTodo = () => {
+    const newTodo: Todo = {
```

Comment:
The 'id' of the newTodo is being set as 'todos.length + 1'. If a todo is deleted, the length will decrease and you may end up with duplicate ids for different todos. Consider using a unique id generator for the 'id' field.
![image](https://github.com/mate-academy/react_todo-app/assets/99403198/16615bb6-e959-4746-825e-42bbaf506e88)